tox.ini

```diff
@@ -23,9 +23,16 @@ commands =
 
 [testenv:lint]
 basepython = python3
+# `pylint` will examine the source code, not the version that would otherwise be
+# installed in `site-packages`, so we use an editable install to make sure the
+# compiled modules are built into a valid place for it to find them.
+package = editable
```

Comment:
Maybe we should do this globally in the tox config. I'm constantly hitting this locally when using tox which is why I pushed https://github.com/Qiskit/qiskit/pull/11380

Reply:
Let's merge #11380 in that separate PR after this, maybe? To me, this change is required to fix the lint job (at the moment, `tox -e lint` only works if you happen to already have built the rust extensions in place yourself, hence the CI failure earlier), but #11380 isn't strictly _needed_, it's a change to the config to deliberately override some of tox's isolation as a trade-off to get better re-test performance.